src/physics/lsm_driver.f90

```diff
@@ -797,9 +824,138 @@ subroutine lsm_init(domain,options)
+            endif
+
+            lake_count=0
+            if(ISLAKE==-1) then
+                if(this_image()==1) write(*,*)  "   WARNING: no lake category in LU data: The model will try to guess lake-gridpoints. This option has not been properly tested!"
+                lakeflag=0  ! If no lake cat is provided, the lake model will determine lakes based
+                            ! on the criterion (ivgtyp(i,j)==iswater .and. ht(i,j)>=lake_min_elev))
+            else
+                lakeflag=1
+                ! from WRF's module_initialize_real.F:
+                DO j = jts, MIN(jde-1,jte)
+                    DO i = its, MIN(ide-1,ite)
+                    !    IF ( grid%lu_index(i,j) .NE. grid%islake ) THEN
+                        if(domain%veg_type(i,j) .NE. ISLAKE ) then
+                            lakemask(i,j) = 0       ! grid%lakemask(i,j) = 0
+                        ELSE
+                            lakemask(i,j) = 1       ! grid%lakemask(i,j) = 1
+                            lake_count= lake_count + 1
+                        end if
+                    END DO
+                END DO
+            endif
+            ! if(options%parameters%debug) write(*,*)"   ",lake_count, " lake cells in image ", this_image()
+
+            ! setlake_depth_flag and use_lakedepth flag. (They seem to be redundant, but whatever):
+            if( associated(domain%lake_depth%data_2d) ) then
+                if(this_image()==1) write(*,*) "   Using Lake depth data "
+                use_lakedepth = 1
+                lake_depth_flag = 1
+            else
+                use_lakedepth = 0
+                lake_depth_flag = 0
+            endif
+
+            call lakeini( &
+                IVGTYP = domain%veg_type                        &
+                ,ISLTYP = domain%soil_type                      &
+                ,HT=domain%terrain%data_2d                      & ! terrain height [m] if ht(i,j)>=lake_min_elev -> lake  (grid%ht in WRF)
+                ,SNOW=domain%snow_water_equivalent%data_2d      & !i  ! SNOW in kg/m^2  (NoahLSM: SNOW liquid water-equivalent snow depth (m)
+                ,lake_min_elev=5.                               & ! minimum elevation of lakes. May be used to determine whether a water point is a lake in the absence of lake category. If the landuse type includes 'lake' (i.e. Modis_lake and USGS_LAKE), this variable is of no effects.
+                ,restart=.false.                                &
+                ,lakedepth_default=50.                          & ! default lake depth (If there is no lake_depth information in the input data, then lake depth is assumed to be 50m)
+                ,lake_depth=domain%lake_depth%data_2d           & !INTENT(IN)
+                ,lakedepth2d=lakedepth2d                        & !INTENT(OUT) (will be equal to lake_depth if lake_depth data is provided in hi-res input, otherwise lakedepth_default)
```

Comment:
does lakedepth2d change through the run?

Reply:
No, lakeini sets it once (depending on lake_depth) and gives it to lake